src/deluge/gui/ui/keyboard/column_controls/chord_mem.cpp

```diff
@@ -64,4 +65,67 @@ void ChordMemColumn::handlePad(ModelStackWithTimelineCounter* modelStackWithTime
 		}
 	}
 };
+
+void ChordMemColumn::writeToFile(Serializer& writer) {
+	int num = 0;
+	for (int32_t y = 0; y < kDisplayHeight; y++) {
+		if (chordMemNoteCount[y] > 0) {
+			num = y + 1;
+		}
+	}
+
+	if (num == 0) {
+		return; // no state to save
+	}
+
+	writer.writeOpeningTag("chordMem");
+	for (int32_t y = 0; y < num; y++) {
+		writer.writeOpeningTag("chordSlot");
```

Comment:
For my PR that saves chordmem to global song, I copied the same logic that you use here. The only thing that I changed was renaming "chordSlot" to just "chord" to make it shorter. Maybe good if you also update the tag to just "chord" so both are equal.

Reply:
Since yours is song level and this is clip level I like the separate names - duplicate names in parsing song vs. clip parameters has caused a few bugs already